fakestorage/response.go

```diff
@@ -19,30 +20,36 @@ type listResponse struct {
 	Prefixes []string      `json:"prefixes"`
 }
 
-func newListBucketsResponse(bucketNames []string) listResponse {
+func newListBucketsResponse(buckets []backend.Bucket) listResponse {
 	resp := listResponse{
 		Kind:  "storage#buckets",
-		Items: make([]interface{}, len(bucketNames)),
+		Items: make([]interface{}, len(buckets)),
 	}
-	sort.Strings(bucketNames)
```

Comment:
we were previously always replying to bucket listings with ordered lists... but i'd say that does not match gcs real behavior, and despite it may facilitate assertions, i think is better to return disordered lists / matching real behavior

Reply:
Makes sense. We can sort in the tests for assertions.